tests/ecmp/test_fgnhg.py

```diff
@@ -150,26 +153,56 @@ def setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts, router_mac, net_p
     return port_list, ip_to_port, bank_0_port, bank_1_port
 
 
+def configure_dut(duthost, cmd):
+    logger.info("Configuring dut with " + cmd)
+    duthost.shell(cmd)
+
+
 def fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, prefix):
 
+    # Init base test params
     if isinstance(ipaddress.ip_network(prefix), ipaddress.IPv4Network):
         ipcmd = "ip route"
     else:
         ipcmd = "ipv6 route"
 
-    for nexthop in ip_to_port:
-        duthost.shell("vtysh -c 'configure terminal' -c '{} {} {}'".format(ipcmd, prefix, nexthop))
-    time.sleep(1)
+    vtysh_base_cmd = "vtysh -c 'configure terminal'"
 
     test_time = str(datetime.now().strftime('%Y-%m-%d-%H:%M:%S'))
 
-    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.create_flows.log".format(test_time)
 
+    ### Start test in state where 1 link is down, when nexthop addition occurs for link which is down, the nexthop
+    ### should not go to active
+    shutdown_link = bank_0_port[0]
+    dut_if_shutdown = ptf_to_dut_port_map[shutdown_link]
+    logger.info("Initialize test by creating flows and checking basic ecmp, "
+                "we start in a state where link " + dut_if_shutdown + " is down")
+
+    configure_dut(duthost, "config interface shutdown " + dut_if_shutdown)
+    time.sleep(30)
+
+    # Now add the route and nhs
+    cmd = vtysh_base_cmd
+    for nexthop in ip_to_port:
+        cmd = cmd + " -c '{} {} {}'".format(ipcmd, prefix, nexthop)
+    configure_dut(duthost, cmd)
+    time.sleep(3)
+
+    # Calculate expected flow counts per port to verify in ptf host
     exp_flow_count = {}
     flows_per_nh = NUM_FLOWS/len(port_list)
     for port in port_list:
         exp_flow_count[port] = flows_per_nh
+        
+    flows_to_redist = exp_flow_count[bank_0_port[0]]
```

Comment:
Can we say "shutdown_link" instead of bank_0_port[0]?

Reply:
good catch, will fix